manager/controlapi/ca_rotation.go

```diff
@@ -104,171 +103,237 @@ func validateExternalCAURL(dialer *net.Dialer, tlsOpts *tls.Config, caURL string
 	if parsed.Scheme != "https" {
 		return errors.New("invalid HTTP scheme")
 	}
-	conn, err := tls.DialWithDialer(dialer, "tcp", getConnectionStringFromHost(parsed.Host), tlsOpts)
+	host, port, err := net.SplitHostPort(parsed.Host)
+	if err != nil {
+		return err
+	}
+	if port == "" {
```

Comment:
Poking a bit more I guess go1.8's [`URL.Hostname()`](https://godoc.org/net/url#URL.Hostname) interprets "http://x:1:2" as a single host with no port, at least `SplitHostPort` will assume the same thing.

And I guess the dialer will fail if it's a weird hostname.